sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c

```diff
@@ -213,6 +333,218 @@ void gnrc_ipv6_nib_rtr_adv_pio_cb(gnrc_netif_t *upstream, const ndp_opt_pi_t *pi
         return;
     }
 
-    _configure_subnets(subnets, upstream, pio);
+#if IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS_SIMPLE)
+    /* if we are the only router on this bus, we can directly choose a prefix */
+    _configure_subnets(subnets, 0, upstream, pio);
+#else
+
+    /* store PIO information for later use */
+    _pio_cache = *pio;
+    _upstream  = upstream;
+
+    /* start advertising by sending timeout message to the server thread */
+    msg_t m = {
+        .type = SERVER_MSG_TYPE_TIMEOUT
+    };
+
+    msg_send(&m, _server_pid);
+#endif /* !IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS_SIMPLE) */
+}
+
+#if !IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS_SIMPLE)
+/**
+ * @brief Check if memory region is set to 0
+ *
+ * @param[in]   The memory array to check
+ * @param[in]   The size of the memory array
+ *
+ * @return  true if all bytes are set to 0
+ */
+static bool _is_empty(const uint8_t *addr, size_t len)
+{
+    for (const uint8_t *end = addr + len; addr != end; ++addr) {
+        if (*addr) {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * @brief Insert a l2 address into the `l2addrs` array
+ *
+ * @param[in]   addr    The l2 address to insert
+ * @param[in]   len     l2 address length
+ *
+ * @return  1 if the address was added to the `l2addrs` array
+ *          0 if the address was already in the array
+ *         -1 if there was no more space in the `l2addrs` array
+ */
+static int _insert(const void *addr, size_t len)
```

Comment:
What is `_insert`ed where?

```suggestion
static int _alloc_l2addr_entry(const void *addr, size_t len)
```

would be better IMHO